src/main/java/de/unistuttgart/informatik/fius/icge/simulation/Simulation.java

```diff
@@ -446,8 +447,23 @@ public void addEntity (Entity entity) {
     /**
      * Return entities
      */
-    public ArrayList<Entity> getEntities() {
+    private ArrayList<Entity> getEntities() {
         return this._entities;
     }
 
+    public void setDelay(int delay) {
+        if (delay < 1) // can not be less than 1, but is available due to nicer layout
+            delay = 1;
+        this._delay = delay;
+
+        // apply delay to each entity
+        for (Entity entity : this.getEntities()) {
+            entity.setDelay(delay);

```

Comment:
Related to a previous comment of mine: I think the simulation speed should be controlled via the number of milliseconds per tick.